backend/routes/userRoutes.js

```diff
@@ -172,16 +193,25 @@ router.get('/recommendedEvents/:userId', async (req, res) => {
   let distanceEventToUser = 0
   let distanceScore = 0
   let totalScore = 0
+  let countOfCommentsInEvent = 0
+  let commentScore = 0
+  let eventAttendance = 0
+  let popularityScore = 0
 
   eventsNear.map((e) => {
+    eventAttendance = 0
+    eventsAttendance.map((a) => {
+      if (a.eventId == e.id) eventAttendance = eventAttendance + 1
+    })
     // Date Score -----------------------------------------------------------------------------------------------------------
     eventDate = new Date(parseInt(e.date.slice(0, 2)) + '/' + parseInt(e.date.slice(3, 5)) + '/' + parseInt(e.date.slice(6, 10)))
     dateDifference =  ( ((eventDate.getTime() - today.getTime()) / 1000) / 604800 ) 
 
-    if (dateDifference !=0 ){
+    if (dateDifference > 0 ){
       dateScore = 1 / dateDifference
     } else{
-      dateScore = 1
+      // This is to not recommend past events to users
+      dateScore = - 1000
```

Comment:
Add date filtering in your db query instead of filtering with scores

Reply:
Done